### ISSUE TITLE
Issues adding permissions over CA

### ISSUE THREAD

**Issue by arth0sz** [PKISolutions/PSPKI#195]
I tried following the examples from the documentation.

```PowerShell
PS C:\Users\Administrator> $ACE = @(New-Object PKI.Security.AccessControl.CertificationAuthorityAccessRule ([Security.Principal.NTAccount]"w.cares"), "ManageCA", "Allow")

New-Object : Cannot find type [PKI.Security.AccessControl.CertificationAuthorityAccessRule]: verify that the assembly
containing this type is loaded.
At line:1 char:10
+ $ACE = @(New-Object PKI.Security.AccessControl.CertificationAuthority ...
+          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidType: (:) [New-Object], PSArgumentException
    + FullyQualifiedErrorId : TypeNotFound,Microsoft.PowerShell.Commands.NewObjectCommand
```

It seems like the type simply does not exist in the module. I dug deep and ended up getting this command to execute without error:

```PowerShell
$ACE = @(New-Object SysadminsLV.PKI.Security.AccessControl.CertSrvAccessRule ([Security.Principal.NTAccount]"w.cares"), "ManageCA", "Allow")
```

However, continuing with the example from the documentation, gives the following error:

```PowerShell
PS C:\Users\Administrator> Get-CertificationAuthority "Authority-CA" | Get-CertificationAuthorityAcl | Add-CertificationAuthorityAcl -AccessControlEntry $ACE | Set-CertificationAuthorityAcl -RestartCA
Add-CertificationAuthorityAcl : A parameter cannot be found that matches parameter name 'AccessControlEntry'.
At line:1 char:107
+ ... horityAcl | Add-CertificationAuthorityAcl -AccessControlEntry $ACE |  ...
+                                               ~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidArgument: (:) [Add-CertificationAuthorityAcl], ParameterBindingException
    + FullyQualifiedErrorId : NamedParameterNotFound,Add-CertificationAuthorityAcl
```

Changing the option -AccessControlEntry to -ACE or -AccessRule executes without errors, but ultimately does nothing. No Manage CA permissions have been added for the particular user. Tried to do the same procedure with a group. No dice.

```PowerShell
Get-CertificationAuthority "Authority-CA" | Get-CertificationAuthorityAcl | Add-CertificationAuthorityAcl -ACE $ACE | Set-CertificationAuthorityAcl -RestartCA
```

Not really sure what would need to be fixed here to make things work.

**Comment by Crypt32** [comment]
> It seems like the type simply does not exist in the module

right, it was moved and mentioned in release notes: https://www.pkisolutions.com/tools/pspki/release-notes-for-pspki-v400/



> ```powershell
> -AccessControlEntry $ACE
> ```

this parameter never existed I believe, its full name is `-AccessRule` and shortcut `-ACE`. Did you get it from examples? If so, can you point to that example, so I can fix docs?

> Changing the option -AccessControlEntry to -ACE or -AccessRule executes without errors, but ultimately does nothing.

I will take a look if there are problems.

**Comment by arth0sz** [comment]
> this parameter never existed I believe, its full name is `-AccessRule` and shortcut `-ACE`. Did you get it from examples? If so, can you point to that example, so I can fix docs?

It was part of the examples here for [Set-CertificationAuthorityAcl](https://www.pkisolutions.com/tools/pspki/set-certificationauthorityacl/). The last lines of Example 1 and Example 3 contain `-AccessControlEntry` as a paramater to `Add-CertificationAuthorityAcl`.

**Comment by Crypt32** [comment]
@arth0sz I see you closed this issue. Is CA permissions issue solved for you? Please advice if I still need to investigate it.

Anyways, let's keep it open, because there still is a doc bug I need to address.

**Comment by Crypt32** [comment]
I did some tests using version 4.0.0 and found that the command is indeed working:
```PowerShell
$ACE = New-Object sysadminslv.PKI.Security.AccessControl.CertSrvAccessRule ([Security.Principal.NTAccount]"testuser")
, "ManageCA", "Allow"
get-ca ca01* | Get-CAACL | Add-CAAccessControlEntry -AccessRule $ace | Set-CAACL -RestartCA
```
After running this command, an account `testuser` is added to CA with requested rights.

This needs a couple of additional tests.
```
$ca = Get-CertificationAuthority "Authority-CA"
$ace = New-Object SysadminsLV.PKI.Security.AccessControl.CertSrvAccessRule ([Security.Principal.NTAccount]"w.cares"), "ManageCA", "Allow"
$acl = $ca | Get-CAACL
# what is here? Ensure that the account being added isn't already there:
$acl.Access
# add ACE:
$acl | Add-CAACL -ACE $ace
# is new entry added (or existing entry updated)?
$acl.Access
# what is visible in UI after saving them:
$acl | Set-CAACL -RestartCA
```

**Comment by Crypt32** [comment]
fixed in v4.2
